src/envoy/http/jwt_auth/jwt_authenticator_test.cc

```diff
@@ -731,6 +757,22 @@ TEST_F(JwtAuthenticatorTest, TestOnDestroy) {
   auth_->onDestroy();
 }
 
+TEST_F(JwtAuthenticatorTest, TestNoForwardPayloadHeader) {
+  // In this config, there is no forward_payload_header
+  SetupConfig(kExampleConfigWithoutForwardPayloadHeader);
+  MockUpstream mock_pubkey(mock_cm_, kPublicKey);
+  auto headers = TestHeaderMapImpl{{"Authorization", "Bearer " + kGoodToken}};
+  MockJwtAuthenticatorCallbacks mock_cb;
+  EXPECT_CALL(mock_cb, onDone(_)).WillOnce(Invoke([](const Status &status) {
+    ASSERT_EQ(status, Status::OK);
+  }));
+  auth_->Verify(headers, &mock_cb);
+
+  // Test when forward_payload_header is not set, the output should still
+  // contain the sec-istio-auth-userinfo header for backward compatibility.
+  EXPECT_TRUE(headers.has("sec-istio-auth-userinfo"));
```

Comment:
you should verify that no extra header is added, not just with this particular location.

Reply:
Done.

Reply:
To be clear, you don't need the EXPECT above (that location has no special meaning after this PR). Checking the size as below is ok, but also verify the key is the original one (i.e "Authorization")  (I also expect current implementation or this test does not clear the origin token)

Reply:
The EXPECT above is needed to check the original output from jwt_auth is preserved, which is for backward compatibility (e.g., not breaking legacy backends using it). Authorization Jwt token has been removed from the headers as JwtRule forward is false.